src/ebay-program-badge/program-badge.tsx

```diff
@@ -0,0 +1,12 @@
+import React, { FC } from 'react'
+import { EbayIcon, EbayIconProps } from '../ebay-icon'
+import { ProgramBadge } from './index'
+
+export type EbayProgramBadgeProps = Omit<EbayIconProps, 'type' | 'name' | 'ref'> & {
+    name: ProgramBadge;
+};
+
+const EbayProgramBadge: FC<EbayProgramBadgeProps> = ({ name, ...rest }) =>
+    <EbayIcon {...rest} type="program-badge" name={name as any} />
```

Comment:
Interesting implementation since it use EbayIcon with the soon-to-be-deprecated `type="program-badge"`. I think the cleaner implementation is (so you don't have to cast name with as any):

1. To have a base icon class which is only available internally. This base class can render both icon and program badge, and used by both EbayIcon and EbayProgramBadge.
2. To have separated code between EbayIcon and EbayProgramBadge, but there will some duplicated code.

For now, I'm fine with your current implementation.